server/src/internalClusterTest/java/org/elasticsearch/search/functionscore/QueryRescorerIT.java

```diff
@@ -983,4 +987,65 @@ private QueryBuilder fieldValueScoreQuery(String scoreField) {
             CombineFunction.REPLACE
         );
     }
+
+    public void testRescoreOnInnerHits() throws Exception {
+        assertAcked(
+            prepareCreate("test").setMapping(
+                jsonBuilder().startObject()
+                    .startObject("_doc")
+                    .startObject("properties")
+                    .startObject("nested_field")
+                    .field("type", "nested")
+                    .startObject("properties")
+                    .startObject("field1")
+                    .field("type", "text")
+                    .endObject()
+                    .endObject()
+                    .endObject()
+                    .endObject()
+                    .endObject()
+                    .endObject()
+            ).setSettings(Settings.builder().put(indexSettings()).put("index.number_of_shards", 1))
+        );
+
+        prepareIndex("test").setId("1").setSource("""
```

Comment:
We can make a test more complex with several documents and `window_size` set up.

Reply:
Absolutely, this is the smallest test I could write to test basic functionality. The goal was to avoid investing too much time in tests until we decide we want to move forward.